File: 1813-maximum-erasure-value/maximum-erasure-value.cpp

class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        set<long long int> a;
        long long int i=0,ans=0,j=0,curr=0,n=nums.size();
        while (j<n){
            if (a.find(nums[j])==a.end()){
                a.insert(nums[j]);
                curr+=nums[j];
                ans = max(ans,curr);
                j++;
            }
            else{
                a.erase(nums[i]);
                curr-=nums[i];
                i++;
            }
        }
        return ans;
    }
};

File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        a= set()
        i=0
        ans=0
        j=0
        n= len(nums)
        while j<n:
            if nums[j] not in a:
                a.add(nums[j])
                ans=max(ans,sum(a))
                j+=1
            else:
                a.remove(nums[i])
                i+=1
        return ans
